app/controllers/type-of-appeal.ts

```diff
@@ -18,27 +19,30 @@ function getTypeOfAppeal(req: Request, res: Response, next: NextFunction) {
   }
 }
 
-function postTypeOfAppeal(req: Request, res: Response, next: NextFunction) {
-  try {
-    const validation = typeOfAppealValidation(req.body);
-    if (validation) {
-      return res.render('appeal-application/type-of-appeal.njk', {
-        types: appealTypes,
-        errors: validation,
-        errorList: Object.values(validation)
-      });
+function postTypeOfAppeal(updateAppealService: UpdateAppealService) {
+  return async (req: Request, res: Response, next: NextFunction) => {
+    try {
+      const validation = typeOfAppealValidation(req.body);
+      if (validation) {
+        return res.render('appeal-application/type-of-appeal.njk', {
+          types: appealTypes,
+          errors: validation,
+          errorList: Object.values(validation)
+        });
+      }
+      req.session.appeal.application.appealType = req.body['appealType'];
```

Comment:
Update session after CCD call, if CCD call has an error we would be updating session without having that change persisted in CCD

Reply:
Agreed and changed